Sources/SwiftStylable/Classes/View/Components/ExtendedButton.swift

```diff
@@ -97,25 +105,29 @@ import UIKit
                     bottom: newValue.bottom,
                     trailing: newValue.right
                 )
-                super.contentEdgeInsets = newValue
             } else {
                 super.contentEdgeInsets = newValue
             }
         }
         get {
-            if #available(iOS 15.0, *), let contentInsets = self.configuration?.contentInsets {
-                return UIEdgeInsets(
-                    top: contentInsets.top,
-                    left: contentInsets.leading,
-                    bottom: contentInsets.bottom,
-                    right: contentInsets.trailing
-                )
+            if #available(iOS 15.0, *) {
+                if let contentInsets = self.configuration?.contentInsets {
+                    return UIEdgeInsets(
+                        top: contentInsets.top,
+                        left: contentInsets.leading,
+                        bottom: contentInsets.bottom,
+                        right: contentInsets.trailing
+                    )
+                } else {
+                    return .zero
+                }
             } else {
                 return super.contentEdgeInsets
             }
         }
     }
     
+    @available(iOS, deprecated: 15, message: "'customItemPlacement' is deprecated since the use of UIButtonConfiguration.")
```

Comment:
Ik denk dat de UIButtonConfiguration niet de volledige vervanging is voor deze functionaliteit. Deze custom placement functies zijn wat uitgebreider dan je hebt in UIButtonConfiguration. Werken ze niet meer sinds iOS15?

Ik zou wel eens een dicussie willen hebben over de button configuratie. Moeten we die meer in stylable bouwen of moet dat bij de gebruiker in de app gaan liggen.

Reply:
Vanaf iOS 15 werkt deze functionaliteit niet meer omdat er 2 functies in `UIButton` niet meer worden aangeroepen:
- `titleRect(forContentRect contentRect: CGRect) -> CGRect`
- `imageRect(forContentRect contentRect: CGRect) -> CGRect`

Ik kan wel even kijken of er een andere manier is om dit alsnog werkend te krijgen. Tot nu toe hebben we dit per app handmatig omgebouwd om de knoppen op de juiste manier te lay-outen, dus vandaar dat ik ze als deprecated wilde markeren. Ook zodat hier in nieuwere projecten geen warnings meer over zouden zijn.

Reply:
Het is prima zo hoor. Het enige wat ik er mee wilde zeggen is het het geen 1 op 1 vervanging is.

Reply:
Nee dat snap ik. Heb nog wat tijd gepakt om te kijken of er echt geen alternatief was om dezelfde functionaliteit te kunnen krijgen met de button configuration. Maar het lijkt er toch echt op van niet helaas.